src/WebJobs.Extensions.CosmosDB/Services/CosmosDBService.cs

```diff
@@ -15,21 +15,33 @@ internal sealed class CosmosDBService : ICosmosDBService, IDisposable
         private bool _isDisposed;
         private DocumentClient _client;
 
-        public CosmosDBService(string connectionString, ConnectionMode? connectionMode, Protocol? protocol)
+        public CosmosDBService(string connectionString, ConnectionMode? connectionMode, Protocol? protocol, string preferredLocations, bool useMultipleWriteLocations)
```

Comment:
Can we change this constructor to take a connectionString and a ConnectionPolicy? Then the factory's job is to build that ConnectionPolicy? I have a feeling this is just going to keep growing over time...

Reply:
Good idea.